src/react/components/_dashboard/TasksCard/IncompleteTasks/ByCategory.js

```diff
@@ -42,12 +43,16 @@ const ByCategory = ({ categories }) => {
               aidYears={category.aidYears}
             />
           );
+        } else if (category.key === 'bCourses' && category.tasks.length > 0) {
+          return <BCoursesTasks key={category.key} tasks={category.tasks} />;
         } else if (category.tasks.length > 0) {
           return <ForCategory key={category.key} category={category} />;
```

Comment:
@developish could this be refactored?

```
} else if (category.tasks.length >0 {
  category.key === 'bCourses' ?
    return <BCoursesTasks key={category.key} tasks={category.tasks} />;
  :
    return <ForCategory key={category.key} category={category} />;
}
```

Reply:
I'd would have disagreed on the small refactor, but I refactored the whole thing to a wrapper conditional on length plus a switch on category.key